tests/api/test_mixin_collisions.py

```
@@ -0,0 +1,246 @@
+    pattern this package already uses. Three things are skipped because they do
+    not run at import time, so none of them can build the cycle:
+
+      * function and method bodies -- a deferred import inside a method is the
+        pattern the Phase 5 preamble prescribes, and flagging it would forbid the
+        one spelling that is allowed;
+      * class bodies, which cannot reach the module under construction usefully
+        anyway;
+      * `if TYPE_CHECKING:` blocks, which never execute.
+    """
+
+    def is_deferred(node):
```

Ruff (0.15.21)

[warning] Missing return type annotation for private function `is_deferred`

(ANN202)